doc/dev/installation.rst

```diff
@@ -123,6 +136,14 @@ following make target from the top level directory:
 
   make all
 
+.. Note::
+
+  To install on macOS:
+
+  .. code-block:: console
+
+    BUILD_QIR_STDLIB_FROM_SRC=ON ENABLE_LLD=OFF make all
```

Comment:
More for visual consistency, but in the above section the two platforms are differentiated in the flowing text, whereas here macOS instructions are placed inside a note box 😅

Reply:
In the requirements section, we had a paragraph about installing dependencies on Debian/Ubuntu so that I just extended that section with installing those deps on macOS. In this section, however, we talk about installing all modules using `make all` no mention of any targeted platforms. I thought it's clearer to add the `make all` configurations for installing on macOS as a note here then. I don't have any strong preference here though and happy to accept any suggestion.

Reply:
Ok no worries, was just a thought 👍

Reply:
This is out of scope for this PR, but with multiple systems, it might be worth switching the structure of the page to group by OS, rather than group by step.

This will help users who just want to copy and paste the commands they need from one place.